compiler/test/input/recursive-equal.gr

```diff
@@ -1,21 +1,24 @@
+# These succeed with `let mut`
 export data Opt<a> = None | Some(a)
 
-let z = box(None)
-z := Some(z)
+let mut z = None
+z = Some(z)
 
-let y = box(None)
+let mut y = None
 
 assert !(z == y)
 
-y := Some(y)
+y = Some(y)
 assert z == y
 
-export data R = Atom | Recurisve(Box<Opt<R>>)
+# These fail with `let mut` because we don't know they shouldn't be unwrapped
+# We'd probably need something like `mut Opt<R>`
+export data R = Atom | Recursive(Box<Opt<R>>)
```

Comment:
Might as well drop this comment here, since you pointed out that this can be accomplished with boxes/mutable record fields. _We_ know that `let mut` is a box under the hood, but for the average user they should think of `let mut` as similar to, say, JavaScript `let`.

```suggestion
export data R = Atom | Recursive(Box<Opt<R>>)
```

Reply:
Alright. I'm going to actually create 2 separate files so I won't accept this right now.